src/codecs/jats/index.ts

```diff
@@ -1375,11 +1383,6 @@ export function decodeReference(
       // Not a chapter so title is <source>
       title = textOrUndefined(child(elem, 'source'))
```

Comment:
Do I understand correctly that  if publication type is `'software', 'web', 'patent'` and has a title derived from `['article-title', 'chapter-title', 'data-title']`, then this line will overwrite that title (when present) with `source`? 

Ideally we would support similar logic fo `'software', 'web', 'patent'` reference types. In other words (for those reference types) if there is one of `['article-title', 'chapter-title', 'data-title']` and a `source`, then the former is decoded as the title and the `source` is used to decoded it as part of 'something' (I'm not too concerned about what  personally - book or periodical - open to your thoughts/preference). 

If there's not one of `['article-title', 'chapter-title', 'data-title']` then the source is the title.

Reply:
Might be best to go over this logic first thing on our call. In my adjustment to the code there should be not alteration, whatsoever, to the logic to determine the title.

The only adjustment to the logic should be that we attempt to set the `isPartOf` if the publication type is other in the same way as if it is journal or preprint.

Reply:
If I understand you correctly then we could treat anything that isn't `publication-type=book` or `publication-type=report` in the same way as journal, preprint or other with regards to the preparation of `isPartOf`?

Reply:
Yeah, I'd be happy with that approach. It wouldn't be semantically correct in the JSON since some of these won't be periodicals (e.g. a software reference for a GitHub repo), but I'm fine with that if you are.

And presumably the `meta.type` could be relied upon by client if there's a need to display the types differently.